rfcs/45-stageless.md

```diff
@@ -392,17 +392,37 @@ trait IntoConfiguredSet {
 
 ### Scheduling errors
 
-There are several reasons an app may error due to improper scheduling. Most are related to graph solvability.
+There are several reasons an app may error due to improper scheduling.
+Most are related to graph solvability. The graph is solved in two different ways.
+The first is the graph where system sets and systems are considered graph nodes. This graph is called the hierarchical graph.
+The edges are checked for errors between systems and system sets at the same nesting level.
+The other graph that is checked is the flattened system graph. In this graph system sets are flattened into connections between only systems.
+So the dependencies on system sets become flattened into dependencies on the systems in the set.
 
-1. Your dependencies contain a loop or cycle.
-2. Your hierarchy contains a loop or cycle.
-3. Your hierarchy has a transitive edge (something is both in some set and not at the same time.)
-4. You called `.in_set` with a system's label.
-5. (User-configurable) You have a dependency between two things that aren't in the same set.
-6. (User-configurable) You referenced an "unknown" label (you forgot to add the system or set somewhere).
-7. (User-configurable) You have two things with conflicting data access and ambiguous order.
+1. A system's dependencies contain a loop or cycle.
+2. The hierarchical graph contains a loop or cycle.
+3. The hierarchical graph has a transitive edge. i.e. there are multiple paths between two sets.

```

Comment:
I'm in agreement here; I don't understand why this is a hard error.

Definitely something worth warning about though.

Reply:
I think it's because the edge breaks the hierarchy. i.e. it's not clear what each level of nesting is anymore. So it makes checking the hierarchy impossible.

Reply:
Also, since transitive edges are not necessary in any case (even between systems) a warning should be emitted even for systems, not only sets.